src/main.cpp

```diff
@@ -1,105 +1,144 @@
-    generator.generate_const_scalar_op("abs", 'r', "r", "std::abs(a)");
-    generator.generate_const_scalar_op("abs", 'i', "i", "std::abs(a)");
-
-    // Complex number manipulation functions.
-    generator.generate_const_scalar_op("complex", 'c', "rr", "primitives::Complex(a, b)");
-    generator.generate_const_scalar_op("polar", 'c', "rr", "std::polar<double>(a, b)");
-    generator.generate_const_scalar_op("real", 'r', "c", "std::real<double>(a)");
-    generator.generate_const_scalar_op("imag", 'r', "c", "std::imag<double>(a)");
-    generator.generate_const_scalar_op("arg", 'r', "c", "std::arg<double>(a)");
-    generator.generate_const_scalar_op("norm", 'r', "c", "std::norm<double>(a)");
-    generator.generate_const_scalar_op("conj", 'c', "c", "std::conj<double>(a)");
+    // Absolute value function
+    generator->generate_const_scalar_op("abs", 'r', "r", "std::abs(a)");
+    generator->generate_const_scalar_op("abs", 'i', "i", "std::abs(a)");
+
+    // Complex number manipulation functions
+    generator->generate_const_scalar_op("complex", 'c', "rr", "primitives::Complex(a, b)");
+    generator->generate_const_scalar_op("polar", 'c', "rr", "std::polar<double>(a, b)");
+    generator->generate_const_scalar_op("real", 'r', "c", "std::real<double>(a)");
+    generator->generate_const_scalar_op("imag", 'r', "c", "std::imag<double>(a)");
+    generator->generate_const_scalar_op("arg", 'r', "c", "std::arg<double>(a)");
+    generator->generate_const_scalar_op("norm", 'r', "c", "std::norm<double>(a)");
+    generator->generate_const_scalar_op("conj", 'c', "c", "std::conj<double>(a)");
+}
+
+
+void generate_v1x_specific_functions(func_gen::Generator* generator) {
+    // Basic scalar arithmetic operators
+    generator->generate_const_scalar_op("operator+", 's', "ss", "a + b");
+    generator->generate_const_scalar_op("operator//", 'i', "ii", "div_floor(a, b)");
+
+    // Bitwise operators
+    generator->generate_const_scalar_op("operator>>>", 'i', "ii", "(int64_t)(((uint64_t)a) >> b)");
+}
+
+
+void test_v1x(const std::string &header_file_path, const std::string &source_file_path) {
+    std::unique_ptr<func_gen::Generator> generator{};
+    generator = std::make_unique<func_gen::V1xGenerator>(header_file_path, source_file_path);
+    generate_common_functions(generator.get());
+    generate_v1x_specific_functions(generator.get());
+}
+
+
+void test_v3x(const std::string &header_file_path, const std::string &source_file_path) {
+    std::unique_ptr<func_gen::Generator> generator{};
+    generator = std::make_unique<func_gen::V3xGenerator>(header_file_path, source_file_path);
+    generate_common_functions(generator.get());
+}
+
+
+int main(int argc, char *argv[]) {
+    // Check command line
+    if (argc != 4) {
+        print_usage();
+        return 1;
+    }
+    
+    // Check version, and test v1x or v3x generators
+    std::string_view version{ argv[3] };
+    if (version == "v1x") {
+        test_v1x(argv[1], argv[2]);
```

Comment:
why do these functions have `test` in the name?

Reply:
Thanks Pablo, that's a good point. I've renamed them to `generate_v1x_files` and `generate_v3x_files`.